rotkehlchen/exchanges/htx.py

```diff
@@ -0,0 +1,191 @@
+            [
+                str(k) + '=' + urllib.parse.quote_plus(str(v))  # need to use the quoted string since cursors have `=` and it breaks signatures  # noqa: E501
+                for k, v in sorted(new_params.items())
+                if v is not None
+            ],
+        )
+        # api describes that strings should be joined using '\n'
+        payload = f'GET\n{host}\n{path}\n{param_str}'
+        dig = hmac.new(
+            key=self.secret,
+            msg=payload.encode('utf-8'),
+            digestmod=hashlib.sha256,
+        ).digest()
+        new_params['Signature'] = base64.b64encode(dig).decode()
+        return new_params
+
+    def _query(self, absolute_path: str) -> Any:
+        url = f'{HUOBI_BASE_URL}{absolute_path}'
+        signed_payload = self._sign_request(url=url, params={})
+        try:
+            response = self.session.get(
+                url=url,
+                params=signed_payload,
+                timeout=CachedSettings().get_timeout_tuple(),
+            )
+            res_body = response.json()
+        except requests.RequestException as e:
+            log.error(f'Failed to query HTX api. Exception: {e}, URL: {url}')
+            raise RemoteError('Failed to query HTX api due to a network or response error.') from e
+
+        if res_body.get('status') != 'ok':
+            log.error(f'Error response from HTX. URL: {url}, Response: {response.text}')
+            raise RemoteError('Received an error response from HTX while querying.')
+
+        if (body := res_body.get('data')) is None:
+            log.error(f'HTX endpoint missing data attribute. URL: {url}, Response: {response.text}')  # noqa: E501
+            raise RemoteError('Requested HTX endpoint does not have a data attribute.')
+
+        return body
+
+    def get_accounts(self) -> list[dict[str, Any]]:
+        """Query the accounts to which the api key has access to"""
+        return self._query(absolute_path='/v1/account/accounts')
+
+    def query_balances(self, **kwargs: Any) -> ExchangeQueryBalances:
+        """Query balances for the accounts linked to the api key"""
+        returned_balances: dict[AssetWithOracles, Balance] = {}
+        for account in self.get_accounts():
+            account_id = account['id']
+            path = f'/v1/account/accounts/{account_id}/balance'
+            try:
+                data = self._query(path)
+            except RemoteError as e:
+                error_prefix = 'Failed to query HTX'
+                log.error(f'{error_prefix} balances due to {e}')
+                return returned_balances, f'{error_prefix} due to a remote error. Check logs for more details'  # noqa: E501
+
+            if (account_balance_type := data['type']) is None:
+                log.error(f'Response for balances does not contain the type key {data}. Skipping')
+                continue
+            if account_balance_type not in self.valid_account_balances_types:
```

Comment:
Empty newline after continue here and below

```suggestion
                continue

            if account_balance_type not in self.valid_account_balances_types:
```